src/main/g8/$if(namespaced.truthy)$src$else$.$endif$/test/$if(namespaced.truthy)$resources$else$.$endif$/$if(namespaced.truthy)$$package$$else$.$endif$/$if(namespaced.truthy)$$name__camel$$else$.$endif$/Test$name__Camel$.tdml

```diff
@@ -26,12 +26,12 @@ limitations under the License.
   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
   defaultRoundTrip="onePass">
 
-  <tdml:parserTestCase name="test_$name;format="camel"$_01" root="$name;format="Camel"$" model="$package;format="packaged"$/$name;format="camel"$/xsd/$name;format="camel"$.dfdl.xsd">
+  <tdml:parserTestCase name="test_$name;format="camel"$_01" root="$name;format="Camel"$" model="$if(namespaced.truthy)$$package;format="packaged"$/$name;format="camel"$/xsd/$endif$$name;format="camel"$.dfdl.xsd">
     <tdml:document>
-      <tdml:documentPart type="file">$package;format="packaged"$/$name;format="camel"$/data/test_01.$name;format="camel"$</tdml:documentPart>
+      <tdml:documentPart type="file">$if(namespaced.truthy)$$package;format="packaged"$/$name;format="camel"$/$endif$data/test_01.$name;format="camel"$</tdml:documentPart>
     </tdml:document>
     <tdml:infoset>
-      <tdml:dfdlInfoset type="file">$package;format="packaged"$/$name;format="camel"$/infosets/test_01.$name;format="camel"$.xml</tdml:dfdlInfoset>
+      <tdml:dfdlInfoset type="file">$if(namespaced.truthy)$$package;format="packaged"$/$name;format="camel"$/$endif$infosets/test_01.$name;format="camel"$.xml</tdml:dfdlInfoset>
```

Comment:
As long as the TDML file, data subdirectory, and infosets subdirectory are all in the same directory, you shouldn't need any namespace directory names.  As I said before, relative-looking filenames are sufficient regardless of which namespace packages the files actually live under:

```
      <tdml:documentPart type="file">data/test_01.$name;format="camel"$</tdml:documentPart>
```

Reply:
> As long as the TDML file, data subdirectory, and infosets subdirectory are all in the same directory, you shouldn't need any namespace directory names.

Ah, I misunderstood what you were saying. I didn't realize Daffodil looked for data/infoset files relative to the TDML file. Good to know, I'll fix that.

Reply:
> Even this tier of "data" and "infoset" subdir is troublesome to new DFDL users. 

I think the main reason for seprate data/infoset directories is that it makes the .gitattributes file very easy--anything in the data dir should have `text=false` so that git doesn't do it's silly line-ending CRLF stuff. For example, we have this:

```
test/data/** text=false
```

I expect that users most likely won't always use `.dat` and will use a specific extension for test data that we can't know about, so we can't use something like this:

```
test/*.dat text=false
```

And I don't think we can expect users to know to update the .gitattributes file for new extensions, I think most people don't even know it exists.

Though, maybe we can do something like this:
```
test/** text=false
test/**/*.tdml text=auto
test/**/*.xml text=auto
```
So everything in the test directory defaults to disabling auto-crlf except for tdml and xml files. This has the downside that if files are added that should have auto-crlf enabled but aren't .tdml or .xml files, then we won't get the expected behavior. But I'm not sure what those files would be, maybe documentation or something? Maybe that's not worth worrying about?

Reply:
Add test/**/*.xsd text=auto as well. test diredtory may also contain schema files that are just for testing purposes.

Reply:
Good call, done

Reply:
Today's changes look good.

I have more to say about data/infosets test subdirectories.  On one hand, I agree new users may find their names confusing and this template doesn't need test subdirectories.  On the other hand, I just added 4 more test files to 68 test files in Daffodil's own [daffodil-test/src/test/resources/org/apache/daffodil/runtime2](https://github.com/apache/daffodil/tree/main/daffodil-test/src/test/resources/org/apache/daffodil/runtime2) directory. I think it would be a great idea to move the 28 *.dat files into a data subdirectory, the 28 *.dat.xml files into an infosets subdirectory, and leave only the 8 *.dfdl.xsd and 8 *.tdml files in the runtime2 directory in a future PR.  I also could replace 28 .dat filenames in Rat.scala with 1 data filename if a directory or a glob expression can be a filename.  Does that change make sense or not?

Reply:
>  I also could replace 28 .dat filenames in Rat.scala with 1 data filename if a directory or a glob expression can be a filename. 

I *think* globs work in the rat excludes list, but I actually kindof like having to manually list files. It's verbose and requires a bit of extra work, but it ensures we know exactly what files are being excluded and that something doesn't slip through the cracks. I think if we standardized on some rule like "all files that cannot include a license and must be excluded must end in .dat.ratexclude" or something, I'd be more fine with just one glob since it's very explicit, but I'm not sure I like that.

Another alternative to avoiding rat excludes is to embed the binary data in the TDML file. Though that does make testing with something like the CLI a bit more difficult though.

Another thought regarding rat exludes, repos created with this template have very different requirements than Daffodil, and generally won't be bound by the Apache license and ASF requirements. In most cases they probably won't even care about running rat. And if they do, they could add a rat excludes for something like "*.ext" if they wanted to glob exclude their test files.